daml-foundations/daml-tools/daml-cli/DA/Cli/Visual.hs

```diff
@@ -74,60 +78,98 @@ darToWorld manifest pkg = AST.initWorldSelf pkgs pkg
     where
         pkgs = map dalfBytesToPakage (dalfsContent manifest)
 
+tplName :: LF.Template -> String
+tplName LF.Template {..} = T.unpack $ head (LF.unTypeConName tplTypeCon)
 
-templateInAction :: Action -> LF.TypeConName
-templateInAction (ACreate  (LF.Qualified _ _ tpl) ) = tpl
-templateInAction (AExercise  (LF.Qualified _ _ tpl) _ ) = tpl
+handlechioceAndAction :: ChoiceAndAction -> LF.ChoiceName
+handlechioceAndAction (ChoiceAndAction tpl (LF.TemplateChoice _ (LF.ChoiceName "Create")  _ _ _ _ _ _) _)  = LF.ChoiceName $ T.pack (tplName tpl ++ "_Create")
```

Comment:
```suggestion
handlechioceAndAction (ChoiceAndAction tpl choice _)
  | LF.chcName choice == LF.ChoiceName "Create" = LF.ChoiceName $ tplName tpl <> "_Create"
```
This operates under the assumption you changed the type of `tplName` as proposed above. You might also need `{-# LANGUAGE OverloadedStrings #-}`.

Reply:
Done :)